File: flight_cheapest_date_search/v1/get/node/flight_cheapest_date_search.js
var Amadeus = require("amadeus");
var amadeus = new Amadeus({
  clientId: 'YOUR_API_KEY',
  clientSecret: 'YOUR_API_SECRET'
});

// Find cheapest dates from Madrid to London
amadeus.shopping.flightDates.get({
  origin: 'MAD',
  destination: 'LON'
}).then(function (response) {
  console.log(response);
}).catch(function (response) {
  console.error(response);
});

File: flight_cheapest_date_search/v1/get/python/flight_cheapest_date_search.py
# Install the Python library from https://pypi.org/project/amadeus
from amadeus import Client, ResponseError

amadeus = Client(
    client_id='YOUR_AMADEUS_API_KEY',
    client_secret='YOUR_AMADEUS_API_SECRET'
)

try:
    '''
    Find cheapest dates from Madrid to London
    '''
    response = amadeus.shopping.flight_dates.get(origin='MAD', destination='LON')
    print(response.data)
except ResponseError as error:
    raise error
